### PR TITLE
Derive arbitrary for rpc `Header` and `Transaction`

### PR DESCRIPTION
**Pull request by tcoratger**
## Solution

- Derive `arbitrary` for `Header` and `Transaction` rpc types.

## PR Checklist

- [X] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types/src/eth/transaction/mod.rs [hunk 1]
```diff
@@ -32,6 +32,7 @@ pub use signature::{Parity, Signature};
 
 /// Transaction object used in RPC
 #[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(all(any(test, feature = "arbitrary"), feature = "std"), derive(arbitrary::Arbitrary))]
```
Comment (DaniPopes) [inline crates/rpc-types/src/eth/transaction/mod.rs:35]: arbitrary should imply std as the dep does not support no_std https://docs.rs/crate/arbitrary/latest/features

**Comment by tcoratger** [comment]
@DaniPopes Is the wasm error linked to the PR or not at all, seems strange no?

File: crates/rpc-types/Cargo.toml [hunk 2]
```diff
@@ -19,7 +19,7 @@ alloy-serde.workspace = true
 alloy-genesis.workspace=true
 
 alloy-consensus = { workspace = true, features = ["serde"] }
-alloy-eips = {workspace = true, features = ["std", "serde"]}
+alloy-eips = {workspace = true, features = ["std", "serde", "arbitrary"]}
```
Comment (DaniPopes) [inline crates/rpc-types/Cargo.toml:22]: arbitrary is optional, should go in the features table

File: crates/eips/src/eip2930.rs [hunk 3]
```diff
@@ -45,8 +46,9 @@ impl AccessListItem {
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Default, RlpDecodableWrapper, RlpEncodableWrapper)]
 #[cfg_attr(
     all(any(test, feature = "arbitrary"), feature = "std"),
```
Comment (DaniPopes) [inline crates/eips/src/eip2930.rs:48]: same here re: std feature

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/rpc-types/Cargo.toml (hunk 2 above)
Comment (tcoratger) [reply crates/rpc-types/Cargo.toml:22]: Fixed

File: crates/eips/src/eip2930.rs (hunk 3 above)
Comment (tcoratger) [reply crates/eips/src/eip2930.rs:48]: Fixed

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**